packages/grid/x-data-grid-pro/src/tests/statePersistence.DataGridPro.test.tsx

```diff
@@ -3,38 +3,99 @@ import {
   DataGridPro,
   DataGridProProps,
   GridApi,
+  GridColDef,
   GridInitialState,
   GridPreferencePanelsValue,
+  GridRowsProp,
   useGridApiRef,
 } from '@mui/x-data-grid-pro';
 import { createRenderer, screen } from '@mui/monorepo/test/utils';
-import { useMovieData } from '@mui/x-data-grid-generator';
 import { expect } from 'chai';
 import { getColumnHeadersTextContent, getColumnValues } from '../../../../../test/utils/helperFn';
 
 const isJSDOM = /jsdom/.test(window.navigator.userAgent);
 
+const rows: GridRowsProp = [
+  { id: 0, category1: 'Cat A' },
+  { id: 1, category1: 'Cat A' },
+  { id: 2, category1: 'Cat A' },
+  { id: 3, category1: 'Cat B' },
+  { id: 4, category1: 'Cat B' },
+  { id: 5, category1: 'Cat B' },
+];
+
+const columns: GridColDef[] = [
+  {
+    field: 'id',
+    type: 'number',
+  },
+  {
+    field: 'category1',
+  },
+  {
+    field: 'idBis',
+    type: 'number',
+    valueGetter: (params) => params.row.id,
+  },
+];
+
+const FULL_INITIAL_STATE: GridInitialState = {
+  columns: {
+    columnVisibilityModel: { category1: false },
+    orderedFields: ['id', '__row_group_by_columns_group__', 'idBis', 'category1'],
+    dimensions: {
+      category1: {
+        width: 75,
+        maxWidth: Infinity,
```

Comment:
```js
JSON.stringify({ a: Infinity }) ))) '{ "a": null }'
```

Indeed
It should be fine here since `Infinity` is the default value for `maxWidth`, so replacing it with `null` works.